### PR TITLE
Resolves conflict from #111

### PR DESCRIPTION
**Pull request by stevempope**
Please see #111 for comment and commit tree. This PR resolves a conflict from the slightly stale branch while adding all of #111 features. The idea is to move the function of the member into the direction of our diagrams.

As we agreed a few weeks ago the idea is to merge often with small features.

### CODE REVIEW

File: mcomp_Java/src/common/interfaces/Membership.java [hunk 1]
```diff
@@ -3,13 +3,26 @@
...
+ * 
+ * They must also be able to use and validate a leaders
+ * public key.
+ * 
  * @author David Avery
+ * @author Stephen Pope 15836791
  *
  */
 public interface Membership {
-  boolean joinHerd(Leadable h);
```
Comment (SamanthaDawsonBanks) [inline mcomp_Java/src/common/interfaces/Membership.java:11]: as discussed we need to consider how the messanging for being moved to a new herd is done but that is to be an enh

File: mcomp_Java/src/common/objects/Member.java [hunk 2]
```diff
@@ -24,119 +25,98 @@
 public class Member implements Rmiable, Membership {
 	private ArrayList<Ability> abilities;
 	private String herdID = null;
-	private String publicKey;
-	private String privateKey;
-	private ArrayList<Member> herdMembers;
```
Comment (SamanthaDawsonBanks) [inline mcomp_Java/src/common/objects/Member.java:29]: equally discussed, member needs a local herd storage object

File: mcomp_Java/src/common/objects/Member.java [hunk 3]
```diff
@@ -24,119 +25,98 @@
...
-		 * Herd object to this method. We need a method that
-		 * allows the Herd to "advertise" itself. The Member
-		 * should be able to respond to this advertisement 
-		 * when appropriate. 
-		 */
-		herdMembers = h.requestJoin(this);
-		if (herdMembers.isEmpty() == true) {
-			System.out.println("Failed to join Herd!");
+	@Override
+	public boolean setPublicKey(Key pk) {
```
Comment (SamanthaDawsonBanks) [inline mcomp_Java/src/common/objects/Member.java:90]: consider renaming this method as it could confuse when pubkey (leader vs self)

**Review by SamanthaDawsonBanks — APPROVED** [review]
approved as an interim, (for leader member merge) but will require some fixes, other than those mentioned; looks good so far :)

File: mcomp_Java/src/common/interfaces/Membership.java (hunk 1 above)
Comment (stevempope) [reply mcomp_Java/src/common/interfaces/Membership.java:11]: This comes to a decision we need to make about how the Leader makes a merge between Herds happen, but I will raise it as an issue and address it in future commits.

**Review by RyanShoobert — APPROVED** [review]

File: mcomp_Java/src/common/objects/Member.java (hunk 2 above)
Comment (stevempope) [reply mcomp_Java/src/common/objects/Member.java:29]: Agreed, will add in again. Discussion needed to be had before I returned it, as again we weren't sure how fault tolerance balances with security for Herd mechanics.

File: mcomp_Java/src/common/objects/Member.java (hunk 3 above)
Comment (stevempope) [reply mcomp_Java/src/common/objects/Member.java:90]: Agreed! I very recently put the leader public key in so this was a bit of an oversight, the only place to set the member key should be in the constructor.

Comment (stevempope) [reply mcomp_Java/src/common/objects/Member.java:90]: Addressed in new branch, will be in next PR.

File: mcomp_Java/src/common/objects/Member.java (hunk 2 above)
Comment (stevempope) [reply mcomp_Java/src/common/objects/Member.java:29]: I have re added these and they will be in the next PR.

**Pull request merged**